lib/widgets/weekday.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:concordia_navigation/models/course.dart';
+import 'package:concordia_navigation/providers/map_data.dart';
+import 'package:concordia_navigation/services/size_config.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:concordia_navigation/storage/app_constants.dart' as constants;
+import 'package:provider/provider.dart';
+
+class Weekday extends StatelessWidget {
+  String weekday;
+  List<Course> courseList;
+
+  Weekday(String weekday, List<Course> courseList) {
+    this.weekday = weekday;
+    this.courseList = courseList;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    DateFormat formatter = DateFormat("Hm");
+
+    List<Widget> courseContainer = [];
+    courseContainer.add(Text(weekday, style: TextStyle(fontSize: 20.0)));
+
+    courseList.forEach((course) => courseContainer.add(ListTile(
+        contentPadding: EdgeInsets.only(
+            left: SizeConfig.safeBlockHorizontal * 5.0,
+            right: SizeConfig.safeBlockHorizontal * 5.0),
+        leading: CircleAvatar(backgroundImage: AssetImage('assets/logo.png')),
+        title: Text(
+          course.summary,
+        ),
+        subtitle: Row(mainAxisSize: MainAxisSize.min, children: <Widget>[
+          Text(
+            "${formatter.format(course?.start?.toLocal())} - ${formatter.format(course?.end?.toLocal())} | ",
+          ),
+          Text((course == null || course?.location == '')
```

Comment:
To add, this will cause an overflow if the Location of the class is too long, one example is `JMSBS1.123`. 
One fix is to wrap it in a `FittedBox`.

Reply:
Is there a way we can make the text smaller to fit the width of its space? I know thats something frequently used in android

Reply:
Yes, use a `FittedBox`

Reply:
Fixed